### PR TITLE
adding evcd support for vcd diff

### PR DESCRIPTION
**Pull request by franout**
hi

i am adding evcd support for the vcdfitool with (i hope) minimal changes in the code

for activating the evcd support i added an additional parameters ``` --extended```

### CODE REVIEW

File: vcddiff.c [hunk 1]
```diff
@@ -1492,6 +1533,8 @@ static void print_help(void)
   printf("\tPrints the current state of variables, instead of\n \tthe default edge value. \n");
   printf(" --wrap  \n -w \n");
   printf("\tWraps the line, only used for default edge print values\n\tnot --state print messages. \n");
+  printf(" --extended  \n -e \n");
+  printf("\tConsider files as extended VCD format.\n\t \n");
```
[the remark this answers is not in the document]
Comment (franout) [reply vcddiff.c:1535]: @wsnyder  woudl you suggest to have a check on the file extesion ? .vcd vs .evcd
together with the checks in the file

[the remark this answers is not in the document]
Comment (wsnyder) [reply vcddiff.c:1535]: No, I personally at least don't name them by any standard, please choose based on file contents.

**Review by wsnyder — COMMENTED** [review]
Great!

For the lines you added, please untabify, and remove trailing spaces.

When I run this with Verilator I get test failures, so something broke, please resolve that.

**Comment by franout** [comment]
@wsnyder  sorry to  reach here so late

> For the lines you added, please untabify, and remove trailing spaces.
will do it 

> When I run this with Verilator I get test failures, so something broke, please resolve that.
can you give me a log?

btw, do you think it may be useful to parse also evcd  with the format "pX 5 5 <0" the two fives are for the signal strenght

**Comment by wsnyder** [comment]
Copy the vcddiff binary into your path then run verilator "make test_regress".

>btw, do you think it may be useful to parse also evcd with the format "pX 5 5 <0" the two fives are for the signal strenght

If the standard says that can happen, then yes.

**Comment by franout** [comment]
@wsnyder just passed all the verilator trace related tests

**Comment by wsnyder** [comment]
The spacing still seemed wrong but it looks like it was horribly inconsistent in the original too.  I just cleaned that up.  So please re-checkout master, copy your new files over the top of the old ones, run "make format", check the diffs, then force push that new branch into this area.  The result should be a small clean set of changes I'll then push.

**Comment by wsnyder** [comment]
BTW if those instructions didn't make sense I can do it for you.

**Comment by franout** [comment]
good to go (i hope)

**Comment by wsnyder** [comment]
I still get verilator test failures, e.g. test_regress/t/t_hier_block_trace_fst.pl

**Comment by wsnyder** [comment]
Thanks!  Merged manually to fix formatting.

**Pull request closed without merging**